FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs

```diff
@@ -77,15 +82,22 @@ public async Task RunAsync(CancellationToken cancellationToken = default)
                 try
                 {
                     job.Status = JobStatus.Running;
+                    job.StartedTime = DateTimeOffset.UtcNow;
                     await _jobExecutor.ExecuteAsync(job, cancellationToken);
                     job.Status = JobStatus.Succeeded;
                     await _jobStore.CompleteJobAsync(job, cancellationToken);
+
+                    await _mediator.Publish(new SuccessfulJobNotification(job), CancellationToken.None);
+                    await _mediator.Publish(new FilterNotification(job), CancellationToken.None);
                 }
                 catch (Exception exception)
                 {
                     job.Status = JobStatus.Failed;
                     job.FailedReason = exception.ToString();
+                    job.FailedException = exception;
+
                     await _jobStore.CompleteJobAsync(job, cancellationToken);
+                    await _mediator.Publish(new FailedJobNotification(job), CancellationToken.None);
```

Comment:
Maybe we can add unit tests for publishing successful or failed notification when processing the job.